cmd/util.go

```diff
@@ -28,30 +30,37 @@ func getPrintableHealthPort(port int) string {
 }
 
 func getKubeConfig(ctx context.Context, namespace string, kubeConfig string) (*kube.ImageUpdaterKubernetesClient, error) {
-	var fullKubeConfigPath string
 	var kubeClient *kube.ImageUpdaterKubernetesClient
-	var err error
 
-	if kubeConfig != "" {
-		fullKubeConfigPath, err = filepath.Abs(kubeConfig)
+	loadingRules := clientcmd.NewDefaultClientConfigLoadingRules()
+	loadingRules.DefaultClientConfig = &clientcmd.DefaultClientConfig
+	loadingRules.ExplicitPath = kubeConfig
+	overrides := clientcmd.ConfigOverrides{}
+	clientConfig := clientcmd.NewInteractiveDeferredLoadingClientConfig(loadingRules, &overrides, os.Stdin)
+
+	config, err := clientConfig.ClientConfig()
+	if err != nil {
+		return nil, err
+	}
+
+	if namespace == "" {
+		namespace, _, err = clientConfig.Namespace()
 		if err != nil {
-			return nil, fmt.Errorf("cannot expand path %s: %v", kubeConfig, err)
+			return nil, err
 		}
 	}
 
-	if fullKubeConfigPath != "" {
-		log.Debugf("Creating Kubernetes client from %s", fullKubeConfigPath)
-	} else {
-		log.Debugf("Creating in-cluster Kubernetes client")
+	clientset, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		return nil, err
 	}
```

Comment:
Wondering if we should move these creation logic back to https://github.com/argoproj-labs/argocd-image-updater/blob/master/pkg/kube/kubernetes.go, which holds some other related functions.

Reply:
That is a good idea. I will move it to the package.